docs/source/deploying-volttron/proxy_web_server.rst

```diff
@@ -0,0 +1,323 @@
+By default, the file can be found at /etc/apache2/sites-enabled/000-default.conf.
+
+An example to use as a starting point for the volttron reverse proxy can be found at
+volttron/scripts/admin/apache-proxy/000-default.conf. Within this file, DocumentRoot and Directory will need to be set.
+Both DocumentRoot and Directory should be set to <path to volttron directory>/volttron/volttron/platform/web/static.
+
+
+.. code-block:: console
+
+    <VirtualHost *:80>
+        # Set DocumentRoot file path to
+        # <path to volttron>/volttron/volttron/platform/web/static
+        # DO NOT include a '/' at the end of the file path!
+        DocumentRoot "<VOLTTRON_ROOT>/volttron/volttron/platform/web/static"
+
+        # Set Directory file path to match DocumentRoot
+        <Directory "<VOLTTRON_ROOT>/volttron/volttron/platform/web/static">
+            Options Indexes FollowSymLinks
+            AllowOverride None
+            Require all granted
+        </Directory>
+
+        # Force redirect of http to https
+        RewriteEngine On
+        RewriteCond %{HTTPS} off
+        RewriteRule ^/+(.*) https://%{HTTP_HOST}%{REQUEST_URI} [R=permanent,L]
+    </VirtualHost>
+
+
+ssl.conf File
+-------------
+
+The ssl.conf contains the proxy, cert, and https configuration details. By default, the file can be found
+at /etc/apache2/mods-enabled/ssl.conf.
+
+An example ssl.conf file is provided with volttron and can be found at volttron/scripts/admin/apache-proxy/ssl.conf.
+The file will need to be modified. The SSLCertificateFile, and SSLCertificateKeyFile paths need to be set to appropriate
+cert and key file path.
+
+Generating self-signed Certs
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The following instructions are based on directions for generating a self-signed certificate
+provided by Ubuntu: `https://ubuntu.com/server/docs/security-certificates <https://ubuntu.com/server/docs/security-certificates>`_.
+
+
+.. warning::
+
+    While we generate a self-signed certificate for the demo, it is not recommended to use a self-signed certificate
+    in a production environment, as this is not a secure practice. Instead use your institution's CA and certificates or a
+    trusted 3rd party, e.g. `Let's Encrypt <https://letsencrypt.org/>`_. This is provided for education and demonstrative
+    purposes only.
+
+First, generate a server key:
+
+.. code-block::
+
+    openssl genrsa -des3 -out server.key 2048
+
+This key will require a pass-phrase on startup of the proxy service which can become inconvenient.
+It is possible to create a version of the key that does not require a pass-phrase, though this is less secure.
+This will be done for this demo, but is not recommended (see warning and link above)
+
+.. code-block::
+    openssl rsa -in server.key -out server.key.insecure
+    mv server.key server.key.secure
+    mv server.key.insecure server.key
+
+Generate a CSR.
```

Comment:
spell out what a CSR is in the first usage.

Reply:
Removed! Now using snakeoil as default example